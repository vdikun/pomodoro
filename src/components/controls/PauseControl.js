import React from 'react';
import { connect } from 'react-redux';
import { toggleTimer } from '../../actions';
import ResponsiveIcon from '../common/ResponsiveIcon';

const PauseButton = ({ isTimerActive, onPress }) => {
    //let buttonText = (isTimerActive) ? "Pause Timer" : "Start Timer";
    let iconName = (isTimerActive) ? "pause-circle" : "play-circle";
    return (
        <ResponsiveIcon iconName={iconName} onPress={() => onPress()} />
    );
};

const mapStateToProps = (state) => {
    return {
      isTimerActive: state.pomodoro.activeTimer,
    }
  };
  
  const mapDispatchToProps = (dispatch) => {
      return { 
          onPress: () => {
              dispatch(toggleTimer());
          }
      }
  };

export default connect(mapStateToProps, mapDispatchToProps)(PauseButton);