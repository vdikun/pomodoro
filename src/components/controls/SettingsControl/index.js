import React from 'react';
import { View } from 'react-native';
import { connect } from 'react-redux';
import styles from './styles';
import ResponsiveIcon, { iconNames } from '../../common/ResponsiveIcon';
import { toggleSettings } from './../../../redux/actions';
import SettingsModal from './SettingsModal';
import pomodoroState from './../../../redux/state';

const SettingsControl = ({ onPress, showSettings }) => {
    if (showSettings) {
        return (
            <View style={styles.settingsIconContainer}>
                <View style={styles.settingsIconContainer}>
                    <ResponsiveIcon iconName={iconNames.SETTINGS} onPress={() => onPress()} /> 
                </View>             
                <SettingsModal/>
            </View>
        );
    } else {
        return null;
    }
};

const mapStateToProps = (state) => {
    return {
      showSettings: (state.pomodoro.pomodoroState == pomodoroState.DISABLED)
    }
};

const mapDispatchToProps = (dispatch) => {
    return {
        onPress: () => {
            dispatch(toggleSettings())
        }
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(SettingsControl);