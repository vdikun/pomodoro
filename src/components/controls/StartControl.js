import React from 'react';
import { connect } from 'react-redux';
import { startPomodoro, stopPomodoro } from '../../actions';
import ResponsiveIcon, { iconNames } from '../common/ResponsiveIcon';
import pomodoroState from '../../reducers/state';

const StartButton = ({ theState, startFn, stopFn }) => {
    let iconName = (theState == pomodoroState.DISABLED) ? iconNames.START : iconNames.STOP;
    let onPress = (theState == pomodoroState.DISABLED) ? startFn : stopFn;
    return (
        <ResponsiveIcon iconName={iconName} onPress={() => onPress()} />
    );
};

const mapStateToProps = (state) => {
    return {
        theState: state.pomodoro.pomodoroState
    }
  };

const mapDispatchToProps = (dispatch) => {
    return { 
        startFn: () => {
            dispatch(startPomodoro());
        },
        stopFn: () => {
            dispatch(stopPomodoro());
        }
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(StartButton);