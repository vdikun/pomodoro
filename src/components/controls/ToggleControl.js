import React from 'react';
import { connect } from 'react-redux';
import { togglePomodoro } from './../../redux/actions';
import ResponsiveIcon, { iconNames } from '../common/ResponsiveIcon';

import pomodoroState from './../../redux/state';

const ToggleButton = ({ onPress, theState }) => {
    if (theState == pomodoroState.DISABLED) {
        return null;
    }
    const iconName = iconNames.TOGGLE; 
    return (
        <ResponsiveIcon iconName={iconName} onPress={() => onPress()} />
    );
};

const mapStateToProps = (state) => {
  return {
    theState: state.pomodoro.pomodoroState
  }
};

const mapDispatchToProps = (dispatch) => {
    return { 
        onPress: () => {
            dispatch(togglePomodoro());
        }
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(ToggleButton);