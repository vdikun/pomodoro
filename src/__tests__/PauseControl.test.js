import React from "react";
import { PauseControl } from "./../components/controls/PauseControl";

import renderer from "react-test-renderer";

test("renders correctly", () => {
  isTimerActive = false;
  onPress = () => console.log("pressed");
  const tree = renderer
    .create(<PauseControl isTimerActive={isTimerActive} onPress={onPress} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
