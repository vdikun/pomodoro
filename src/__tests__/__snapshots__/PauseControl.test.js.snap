// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "margin": 5,
      "opacity": 0.7,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 40,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    ï…„
  </Text>
</View>
`;
